internal/audit/kafka/tls.go

```diff
@@ -0,0 +1,96 @@
+package kafka
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+)
+
+const DefaultTLSVersion = tls.VersionTLS12
+
+type TLSReloader struct {
+	mu       sync.RWMutex
+	certPath string
+	keyPath  string
+	cert     *tls.Certificate
+}
+
+func NewTLSReloader(certPath, keyPath string) (*TLSReloader, error) {
+	reloader := &TLSReloader{
+		certPath: certPath,
+		keyPath:  keyPath,
+	}
+
+	cert, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load TLS key pair: %w", err)
+	}
+	reloader.cert = &cert
+
+	go func() {
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, syscall.SIGHUP)
+		for range c {
+			if err := reloader.reload(); err != nil {
```

Comment:
What about instead using a timer that I can configure how often the certificate is reloaded? 
Tools like cert-manager will reload the certificates in-place on disk.

Reply:
https://github.com/cerbos/cerbos/pull/1667/commits/c6138adac080eca256afeede1a71be20085fe739 lmk what you think 👍